### ISSUE TITLE
Cannot have Unicode string

### ISSUE THREAD

**Issue by arjunaskykok** [vyperlang/vyper#2338]
### Version Information

* vyper Version (output of `vyper --version`): 0.2.8+commit.069936f
* OS: linux
* Python Version (output of `python --version`): 3.9.0

### What's your issue about?

This is the Vyper file:
```python
symbol: String[2] = "Ͼ"
```

```bash
$ vyper /tmp/a.vy
Error compiling: /tmp/a.vy
vyper.exceptions.InvalidLiteral: 'Ͼ' is not an allowed string literal character
  contract "/tmp/a.vy", line 1:20
  ---> 1 symbol: String[2] = "Ͼ"
  ---------------------------^
```

### How can it be fixed?

@iamdefinitelyahuman I'm curious why you limited the string to be ascii character. This is the commit: https://github.com/vyperlang/vyper/commit/b623c12eb2d325883244223fa418c1c539b01335

This is the code:
https://github.com/vyperlang/vyper/blob/master/vyper/ast/nodes.py#L791

I'm translating the CryptoPunk Solidity code to Vyper code:
https://github.com/larvalabs/cryptopunks/blob/master/contracts/CryptoPunksMarket.sol

They use Ͼ as the symbol. I know I can use something else, but I'm just curious. Who knows I want to use Japanese string for the symbol of my own token later. :)

**Comment by iamdefinitelyahuman** [comment]
There is some discussion around this on https://github.com/vyperlang/vyper/issues/358.  I believe the choice of ASCII was partly for gas savings, partly to ensure consistency with Solidity.  The code you reference is using an old Solidity version, starting from `v0.7.0` it would no longer compile due to Ͼ being an invalid character.

**Comment by fubuloubu** [comment]
Yeah, part of what is weird about Unicode is that it can take a non-constant amount of bytes per character. All Ethereum tooling, including Solidity, does not support unicode strings, it would probably take a concerted effort to support ecosystem-wide (think Python 2 -> 3)

Vyper could very easily support it because Python has native unicode handling in strings, but the rest of the ecosystem is holding us back.

**Comment by arjunaskykok** [comment]
Solidity already supports Unicode string.

```solidity
pragma solidity ^0.8.1;

contract HelloWorld {
    string public greet = unicode"こんにちは 😃";
}
```

Kinda reminds me of Python 2 ironically.

**Comment by arjunaskykok** [comment]
So Solidity has two kinds of strings to avoid homograph attack.

https://twitter.com/smpalladino/status/1394641266995077122

**Comment by fubuloubu** [comment]
As axic noted, tooling broadly does not support unicode characters like that

We could probably pretty easily support it by converting unicode to bytes, I think ascii chars should take same size by default (e.g. `"stuff" == u"stuff"`)
